modules/core/src/lapack.cpp

```diff
@@ -1396,6 +1405,47 @@ bool cv::eigen( InputArray _src, OutputArray _evals, OutputArray _evects )
         v = _evects.getMat();
     }
 
+#ifdef HAVE_EIGEN
```

Comment:
If some eigen versions doesn't pass accuracy checks then you should avoid using of them here:
```
#if defined HAVE_EIGEN && \
    (EIGEN_MAJOR_VERSION > 3 || (EIGEN_MAJOR_VERSION == 3 && EIGEN_MINOR_VERSION >= 2))
```

Reply:
When I made the changes, there was no Core_PCA test. With Eigen 3.3.2, this branch now passes all Core_Eigen tests but fail in Core_PCA.

So unless we increase the error threshold, there is no way to merge this for now.

Reply:
Old code uses double type for internal calculations. Currently there are two code branches: `float` and `double`. Probably we should increase tolerance for float-based calculations or use double-only code.

Reply:
Alright. Please let me know if you decide to increase the tolerance. I don't think there is anything that I can do before that. Thanks